packages/leaderboard/src/board.ts

```diff
@@ -147,18 +144,28 @@ async function readAssoc(lst: string[]) {
   const web3 = new Web3(LEADERBOARD_WEB3)
   const kit: ContractKit = newKitFromWeb3(web3)
   const accounts: AccountsWrapper = await kit.contracts.getAccounts()
-  lst.forEach(async (a) => {
-    try {
-      const url = await accounts.getMetadataURL(a)
-      console.log(a, 'has url', url)
-      let metadata: IdentityMetadataWrapper
-      if (url == '') metadata = IdentityMetadataWrapper.fromEmpty(a)
-      else metadata = await IdentityMetadataWrapper.fetchFromURL(url)
-      processClaims(kit, a, metadata)
-    } catch (err) {
-      console.error('Bad address', a, err.toString())
-    }
-  })
+  await Promise.all(
+    lst.map(async (a) => {
+      try {
+        const url = await accounts.getMetadataURL(a)
+        console.log(a, 'has url', url)
+        let metadata: IdentityMetadataWrapper
+        if (url == '') metadata = IdentityMetadataWrapper.fromEmpty(a)
+        else {
+          try {
+            metadata = await IdentityMetadataWrapper.fetchFromURL(url)
+          } catch (err) {
+            console.error('Error reading metadata', a, err.toString())
+            metadata = IdentityMetadataWrapper.fromEmpty(a)
+          }
+        }
+        await processClaims(kit, a, metadata)
+      } catch (err) {
+        console.error('Bad address', a, err.toString())
+      }
+    })
+  )
+  client.end()
```

Comment:
Do you think it would be more clear if client.connect and client.end were in the same function?

Reply:
This won't be very clear without using promises.